### PR TITLE
Adding Docker file and updating README

### PR DESCRIPTION
**Pull request by vandanabhandari**
**Objective**
Adding Docker file for the ease of setup. Also added steps for developers who want to deploy in Fargate. I was trying to set up the project to check what can be contributed and faced some challenges because of new Mac OS possibly. With this addition, developers can hop on working on the project without much worrying about setup.

**Motivation**

1. Makes it easy for development setup for new projects using the open source. Developers need not worry about maven and java installations and to have the right version. Often Developers are working on more than one project and dont' want to deal with switching versions based on the project. For example, my mac was new and had java 15 and did not have maven. Maven was failing on main elide project because there was enforcement to use java 8. So I had to brew install jdk8 and use this. This motivated me to add dockerization for all the modules so other developers don't need to run into this issue. This will also help in environments with multiple projects and to containerize.

2. Also added advantage is containerization will help in scaling independently based on the task. I used aws ECR for container registry and Fargate to manage the tasks containers. You could use other registry services like jfrog and container service like Kubernetes.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by karmakoder — COMMENTED** [review]
👍 💥

File: README.md [hunk 1]
```diff
@@ -18,6 +18,40 @@ To run from Heroku:
...
+   docker build -t elide/elide-spring-boot-example .
+2. Run docker container locally
+   docker run -p 8081:8080 elide/elide-spring-boot-example
+   https://localhost:8081/
+   
+# Run from cloud (AWS)
+
+
+1. ECR 
+   i. Create a repository in Elastic Container Registry with name elide-spring-boot-example
```
Comment (moizarafat) [inline README.md:35]: you may want to add linebreaks after each line else all of this will be in 1 single line.

File: README.md [hunk 2]
```diff
@@ -18,6 +18,40 @@ To run from Heroku:
...
 
+## Docker and Containerize
+
+To containerize and run elide project locally
+
+1. Build Docker Image from Root of the project 
+   docker build -t elide/elide-spring-boot-example .
+2. Run docker container locally
+   docker run -p 8081:8080 elide/elide-spring-boot-example
+   https://localhost:8081/
```
Comment (aklish) [inline README.md:29]: Should be http instead of https.

File: README.md (hunk 1 above)
Comment (aklish) [reply README.md:35]: Agreed.  Markdown sub-bullets here would be more readable.

**Review by aklish — CHANGES_REQUESTED** [review]
Thanks for the submission!  This will be really useful.  Added a few comments.

File: README.md [hunk 3]
```diff
@@ -18,6 +18,40 @@ To run from Heroku:
 
 [![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy?template=https://github.com/yahoo/elide-spring-boot-example)
 
+## Docker and Containerize
+
+To containerize and run elide project locally
+
+1. Build Docker Image from Root of the project
```
[the remark this answers is not in the document]
Comment (vandanabhandari) [reply README.md:25]: done

File: README.md (hunk 1 above)
Comment (vandanabhandari) [reply README.md]: done, addressed this in the latest commit.

File: README.md (hunk 2 above)
Comment (vandanabhandari) [reply README.md]: updated the doc to have http://localhost:8081

**Comment by vandanabhandari** [comment]
> Thanks for the submission! This will be really useful. Added a few comments.

**Comment by vandanabhandari** [comment]
Addressed the review comments

**Comment by vandanabhandari** [comment]
@aklish I have addressed the comments, but it still shows requested changes, possibly because its outdated now.

**Review by aklish — APPROVED** [review]

**Pull request merged**